src/Data/Nat.agda

```diff
@@ -28,6 +38,35 @@ open import Data.Nat.Properties public
   ; _<‴?_; _≤‴?_; _≥‴?_; _>‴?_
   )
 
+------------------------------------------------------------------------
+-- Reading a number
+
+readMaybe : ∀ base {base≤16 : True (base ≤? 16)} → String → Maybe ℕ
```

Comment:
I feel that this should live in `Data.Nat.Show` rather than here. I'm not sure we really want to add these dependencies,  for example `Data.List.Categorical`, to `Data.Nat`?

Reply:
I like the idea of `Data.Nat.Show`, and other such `.Show` sub-modules.